### PR TITLE
[vcpkg]Extend vcpkg copy tool dependencies

### PR DESCRIPTION
**Pull request by longhuan2018**
When I wanted to copy all the osg dependencies to the tool directory, I found that vcpkg copy tool dependencies couldn't fulfill all my needs.
For example, it cannot copy the dependencies of the debug library; it cannot copy the dependencies of the dlls in the osgPlugins directory to its parent directory.
So in order to meet these needs, I think it is necessary to make vcpkg copy tool dependencies add the following features:
- [x] Support debug library
- [x] Support user-defined dependency library output path
- [x] Support custom search path for dependent libraries

### CODE REVIEW

**Comment by Neumann-A** [comment]
> Support debug library

There is no reason to install debug tools and as such there is no reason to install debug binaries

**Comment by longhuan2018** [comment]
> > Support debug library
> 
> There is no reason to install debug tools and as such there is no reason to install debug binaries

Well, as long as it can support the last two customizations, it can also be used to implement my copy of the debug dependency requirements.

**Comment by JackBoosY** [comment]
Hi @strega-nil. could you please review this PR?

Thanks.

File: scripts/cmake/vcpkg_copy_tool_dependencies.cmake [hunk 1]
```diff
@@ -16,17 +16,91 @@
...
+        else()
+            list(APPEND _ctdb_SEARCH_DIRS ${CURRENT_PACKAGES_DIR}/debug/bin)
+            list(APPEND _ctdb_SEARCH_DIRS ${CURRENT_INSTALLED_DIR}/debug/bin)
+        endif()
+    endif()
+
+    message(STATUS "vcpkg_copy_tool_dependencies TOOL_DIR = ${_ctdb_TOOL_DIR}" " OUTPUT_DIR = ${_ctdb_OUTPUT_DIR}" " SEARCH_DIRS = ${_ctdb_SEARCH_DIRS}")
     macro(search_for_dependencies PATH_TO_SEARCH)
-        file(GLOB TOOLS ${TOOL_DIR}/*.exe ${TOOL_DIR}/*.dll)
+        #message(STATUS "search_for_dependencies ${PATH_TO_SEARCH}")
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_copy_tool_dependencies.cmake:91]: Could you either leave this in as a diagnostic message to the user, or remove it completely?

File: scripts/buildsystems/msbuild/applocal.ps1 [hunk 2]
```diff
@@ -38,6 +38,11 @@ try
 {
     $baseBinaryPath = Resolve-Path $targetBinary -erroraction stop
     $baseTargetBinaryDir = Split-Path $baseBinaryPath -parent
+    if ($outputDir) {
```
Comment (strega-nil) [inline scripts/buildsystems/msbuild/applocal.ps1:41]: Could you use `[String]::IsNullOrEmpty` here?

File: scripts/buildsystems/msbuild/applocal.ps1 [hunk 3]
```diff
@@ -38,6 +38,11 @@ try
...
+        if (Test-Path "$outputDir\$_"){
+            $baseTargetBinaryDir = $outputDir
```
Comment (strega-nil) [inline scripts/buildsystems/msbuild/applocal.ps1:43]: This should probably `Write-Error` if this check fails.

**Review by strega-nil — CHANGES_REQUESTED** [review]
Other than these small changes, this looks good! Thanks @longhuan2018

**Comment by ras0219-msft** [comment]
Closing for inactivity; thank you for the PR!

**Pull request closed without merging**